ietf/doc/views_draft.py

```diff
@@ -2075,27 +2075,44 @@ def clean(self):
         return cleaned_data
 
 @login_required
-def issue_wg_call_for_adoption(request, name):
+def issue_wg_call_for_adoption(request, name, acronym):
     doc = get_object_or_404(Document, name=name)
-
-    if doc.stream_id != "ietf":
-        raise Http404
-    if doc.group is None or doc.group.type_id != "wg":
-        raise Http404
-    if not can_issue_ietf_call_for_adoption(doc):
-        raise Http404
-
-    if not is_authorized_in_doc_stream(request.user, doc):
-        permission_denied(request, "You don't have permission to access this page.")
+    group = Group.objects.filter(acronym=acronym, type_id="wg").first()
+    if any(
+        [
+            doc.type_id != "draft",
+            group is None,
+            not is_doc_ietf_adoptable(doc),
+            doc.group.acronym == "none" and acronym is None,
+            doc.group.acronym == "none" and not can_adopt_draft(request.user, doc),
+            doc.group.acronym != "none"
+            and not is_authorized_in_doc_stream(request.user, doc),
```

Comment:
The doc may not be in the "ietf" stream at this point - is there a check that the user is authorized to issue the adoption in that stream? It looks to me like it checks its current stream, but that might be a different stream. Am I missing something?